[Impact] List view

AdrienClairembault opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?  | yes

cedric-anne commented on front/impactcsv.php:
```diff
@@ -0,0 +1,100 @@
+// Output csv data
+$filename = $item->fields['name'];
+header("Content-Type: application/csv");
+header("Content-Disposition: attachment; filename='$filename.csv';");
```
It would be more secure to use encoded filename (in `filename*=` parameter of the header) as it cames from the name field of an asset.
For backward compatibility for old browsers, you can keep a `filename="impact.csv";` that will be used by browsers that does not uses the `filename*` parameter.
See https://tools.ietf.org/html/rfc6266#section-5
```suggestion
header("Content-Disposition: attachment; filename="impact.csv"; filename*=UTF-8''".rawurlencode($filename . '.csv'));
```

trasher commented on front/impactcsv.php:
```diff
@@ -0,0 +1,100 @@
+
+// Output csv data
+$filename = rawurlencode("{$item->fields['name']}.csv");
+header("Content-Type: application/csv");
```
Use `text/csv`

AdrienClairembault replied:
Fixed in https://github.com/glpi-project/glpi/pull/6441/commits/523f48d086ab8bb4b55a9169db1127e15e41af5c.

trasher commented on front/impactitem.form.php:
```diff
@@ -0,0 +1,52 @@
+   $id = $_POST["id"] ?? 0;
+
+   // Can't update, id is missing
+   if ($id === 0) {
```
Maybe add an entry in logs

AdrienClairembault replied:
Fixed in https://github.com/glpi-project/glpi/pull/6441/commits/8c10d23585c24b64a9aec8584df6057e0b072b9d.

trasher commented on front/impact.php:
```diff
@@ -40,9 +40,15 @@
+   $graph = Impact::makeDataForCytoscape(Impact::buildGraph($item));
+   $params = Impact::prepareParams($item);
+
+   echo '<div id="impact_graph_view">';
```
Does not `Impcat::displayGraphView()` does the same?

AdrienClairembault replied:
Fixed in https://github.com/glpi-project/glpi/pull/6441/commits/2378c31b317d6b97fb3be66133e964acea9add75.

trasher commented on inc/impact.class.php:
```diff
@@ -127,16 +135,488 @@ public static function displayTabContentForItem(
+         ]) . '</td>';
+         echo '</tr>';
+         echo '<tr>';
+         echo '<td><label>' . __("No limit") . '</label></td>';
```
Lables tags must have a `for` attribute corresponding to the `id` of the form element

AdrienClairembault replied:
Fixed in https://github.com/glpi-project/glpi/pull/6441/commits/8c9164fb530c4497f9756205d9c687e64a1eb3da.

trasher commented on inc/impact.class.php:
```diff
@@ -127,16 +135,488 @@ public static function displayTabContentForItem(
+      // Settings dialog
+      if ($can_update) {
+         echo '<div id="list_depth_dialog" class="impact-dialog" title=' . __("Settings") . '>';
+         echo '<form action="../front/impactitem.form.php" method="POST">';
```
Why not use existing method here instead of hardcoded `form` tag?

AdrienClairembault commented:
@cedric-anne 
> List view seems to be loaded on impact tab display. It means that all impacts added on the graph view are not available until page is refreshed.

The list view is now updated automatically when the graph is saved.

> When saving settings on list view, page is reloaded and graph view is displayed. I would expect to stay on view I was when saving settings.

Fixed, refreshing also keep you on the same view now.

> Rendering seems broken.  

Can you delete your scss cache ? I've found an issue with the cache, it will only check the hash of the styles.scsse and since the impact style is imported using an @import the hash won't change.
Should we improve the scss cache to also check for each @import hash too ?

cedric-anne commented:
> > Rendering seems broken.
> 
> Can you delete your scss cache ? I've found an issue with the cache, it will only check the hash of the styles.scsse and since the impact style is imported using an @import the hash won't change.
> Should we improve the scss cache to also check for each @import hash too ?

Done in https://github.com/glpi-project/glpi/pull/6482

orthagh commented:
some ui issues:

- reduce `.impact-left` to 5-10px padding left
- H3 (impact, impacted by) should be bigger or have margin reduced
- Empty H3 should not be displayed
- icons (right and top ones ) are too big currently
- icons for toggling lists are also too big, reduce to 1.5em and reduce right margin
- don't use th for theses last icons and place them as `float:right` (or flex)

AdrienClairembault commented:
UI changes + rebase done.

orthagh commented:
some last remarks:

- invert the two icons to match style of others glpi viz
- When on load you display list view and switch after to viz mode, grid is not present

orthagh commented:
When no dependencies declared on an item, the list view is empty and we can think it's bugged.

So add a message to inform the user we don't have dependencies yet
